bears/general/QuotesBear.py

```diff
@@ -34,9 +35,15 @@ def correct_single_line_str(self, filename, file, sourcerange,
         str_contents = file[sourcerange.start.line - 1][
                        sourcerange.start.column:sourcerange.end.column-1]
 
-        if preferred_quotation in str_contents:
+        if (preferred_quotation in str_contents and
+                not force_preferred_quotation):
             return
 
+        elif preferred_quotation in str_contents:
+            add_escape = '\\'+preferred_quotation
+            str_contents = str_contents.replace(preferred_quotation,
+                                                add_escape)
```

Comment:
I think you don't need the if, `replace` won't replace if there are no `preferred_quotation` inside `str_contents` ;)

Also I would inline `add_escape`:
```
str_contents = str_contents.replace(preferred_quotation, '\\' + preferred_quotation)
```
or you could actually use `coala_utils.string_process.escape` :) But this is optional, though this function is designed for such scenarios :)

Another thing: It may be worth to add a comment here explaining what this snippet does: Escaping quotation inside the string that would conflict with the preferred quotation used to surround the string :+1:

Reply:
Yeah about the comments i was thinking it to make a newcomer issue ;P

Reply:
and making add_escape inline as in `str_contents.replace(preferred_quotation,add_escape)` ?